pkg/chartRepo/ChartRepositoryService.go

```diff
@@ -172,13 +172,15 @@ func (impl *ChartRepositoryServiceImpl) CreateChartRepo(request *ChartRepoDto) (
 			if err != nil {
 				return nil, err
 			}
-			data, err := impl.updateRepoData(cm.Data, request)
-			if err != nil {
-				impl.logger.Warnw(" config map update failed", "err", err)
-				continue
+			if cm.Data != nil {
+				data, err := impl.updateRepoData(cm.Data, request)
+				if err != nil {
+					impl.logger.Warnw(" config map update failed", "err", err)
+					continue
+				}
+				cm.Data = data
+				_, err = impl.K8sUtil.UpdateConfigMap(impl.aCDAuthConfig.ACDConfigMapNamespace, cm, client)
 			}
-			cm.Data = data
-			_, err = impl.K8sUtil.UpdateConfigMap(impl.aCDAuthConfig.ACDConfigMapNamespace, cm, client)
 		} else {
 			secretLabel := make(map[string]string)
```

Comment:
why diff treatment for private and public repo? can we have an unified mechanism ?

Reply:
Actually argo-cd creates secret kind object for every helm chart, so for private chart repos there is password, and username also which is a part of that manifest so in this way a private chart is being registered by argocd. But for public chart also argocd creates this secret kind object so ideally we should add both type of charts public and private via creating secet kind object which can be done in another ticket.